### DISCUSSION TITLE
MySQL --login-path credentials

### DISCUSSION THREAD

**Discussion by timboss** [mithrandyr/SimplySql#110] [General]
MySQL allows credentials to be saved in the users %appdata%\MySQL\.mylogin.cnf via mysql_config_editor.exe 

Allowing mysql queries to be called using the mysql.exe --login-path paramater

Can this functionality be included for providing credentials in Open-MySqlConnection ? 

Cheers

**Comment by mithrandyr** [comment]
@timboss  -- SimplySQL using the MySql.Data nuget package provided by Oracle.  Creating the connection to a MySQL database happens through this [MySqlConnection object](https://dev.mysql.com/doc/dev/connector-net/8.0/api/data_api/MySql.Data.MySqlClient.MySqlConnection.html).  So this constrains what I can do with the Open-MySqlConnection cmdlet.

**Comment by timboss** [comment]
thanks for the info..

sorted it out using my_print_defaults (
https://www.percona.com/blog/get-passwords-plain-text-mylogin-cnf/)

could this be included in your package?

T

**Reply by mithrandyr** [reply]
@timboss  -- unfortunately this is not feasible.  SimplySql makes no assumptions about what is included on the system -- everything it needs is already included.  That post makes use of a tool included in the default MySql installation, however SimplySql will work without those tools installed.

Moreover, keeping with the nature of Powershell, a module or cmdlet should be focused on doing one thing well.  Extracting the saved credentials that MySql editor persists would properly be the responsibility of a new cmdlet -- but that cmdlet would not fit within the scope and purpose of SimplySql.

I would encourage you to try your hand at creating a cmdlet or function that does this and generates a PSCredential object -- which then could be passed to `Open-MySqlConnection` via `-Credential`
